actionlib/scripts/axclient.py

```diff
@@ -0,0 +1,61 @@
+#! /usr/bin/python
+# **********************************************************
+#  Software License Agreement (BSD License)
+#
+#   Copyright (c) 2019, Open Source Robotics Foundation, Inc.
+#   All rights reserved.
+#
+#   Redistribution and use in source and binary forms, with or without
+#   modification, are permitted provided that the following conditions
+#   are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above
+#      copyright notice, this list of conditions and the following
+#      disclaimer in the documentation and/or other materials provided
+#      with the distribution.
+#    * Neither the name of Willow Garage, Inc. nor the names of its
+#      contributors may be used to endorse or promote products derived
+#      from this software without specific prior written permission.
+#
+#   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+#   FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+#   COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+#   INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+#   BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+#   LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#   CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+#   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+#   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#   POSSIBILITY OF SUCH DAMAGE.
+#
+#  Author: Matt Reynolds
+# **********************************************************
+"""
+Deprecated: Use 'axclient.py' from the 'actionlib_tools' package instead
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+
+import rospkg
+
+
+def main():
+    if "actionlib_tools" in rospkg.RosPack().list():
+        print("Warning: The 'axclient.py' script has been moved to the 'actionlib_tools' package." +
+              " A future version of 'actionlib' will not include this script\n")
+        return os.system(" ".join(["rosrun", "actionlib_tools", "axclient.py"] + sys.argv[1:]))
```

Comment:
Doesn't really matter for something so trivial, but you can save the memory cost of an outer process hanging around in these cases by using one of the `os.exec*` family of functions: https://docs.python.org/3/library/os.html#os.execl

OTOH, I probably would have implemented this as a bash script instead, but really whatever. :)

Reply:
Didn't know those functions existed, thanks! I was unhappy with the extra shell, glad to know there's a solution.

I stuck with python scripts to keep things consistent with the previous `axclient.py` and `axserver.py` scripts, since these are supposed to be shim scripts. I suppose there could be a bash shebang in a `.py` file, but I'm not particularly keen on that :)

Reply:
Oh right yeah that would be pretty sketchy— good call.

Reply:
Switched to `os.exec*` in 3633b69